### ISSUE TITLE
Cannot install v.0.311.0

### ISSUE THREAD

**Issue by tsunheang** [facebook/nuclide#1547]
### Issue and Steps to Reproduce
Describe your issue and tell us how to reproduce it (include screenshots and/or any console messages).

### Expected Behavior
Tell us what should happen.

### Actual Behavior
Tell us what happens instead.

### Versions
* Atom:
* Nuclide:
* Client OS:
* Server OS (optional):

### Additional Details
* Installed packages (`apm ls --installed`):

**Comment by Emperio** [comment]
Installing “nuclide@0.311.0” failed.Hide output…

> runas@3.1.1 install /private/var/folders/f8/7tv3prld4gq7mptx0w375sbm0000gn/T/apm-install-dir-11857-22107-1p8he39.bynspjh5mi/node_modules/nuclide/node_modules/runas
> node-gyp rebuild

/private/var/folders/f8/7tv3prld4gq7mptx0w375sbm0000gn/T/apm-install-dir-11857-22107-1p8he39.bynspjh5mi
`-- (empty)


npm WARN excluding symbolic link pkg/nuclide-fs-atom/__mocks__/fixtures -> ../../nuclide-fs/spec/fixtures
npm WARN excluding symbolic link pkg/nuclide-fs/__mocks__/fixtures/dir/Directory/LinkDotDotSlashTextFile.txt -> ../TextFile.txt
npm WARN excluding symbolic link pkg/nuclide-fs/__mocks__/fixtures/dir/LinkDirectory -> Directory
npm WARN excluding symbolic link pkg/nuclide-fs/__mocks__/fixtures/dir/LinkDirectorySlashTextFile.txt -> Directory/TextFile.txt
npm WARN excluding symbolic link pkg/nuclide-fs/__mocks__/fixtures/dir/LinkLinkDirectory -> LinkDirectory
npm WARN excluding symbolic link pkg/nuclide-fs/__mocks__/fixtures/dir/LinkLinkDirectorySlashTextFile.txt -> LinkDirectorySlashTextFile.txt
npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm WARN excluding symbolic link pkg/nuclide-fs-atom/__mocks__/fixtures -> ../../nuclide-fs/spec/fixtures
npm WARN excluding symbolic link pkg/nuclide-fs/__mocks__/fixtures/dir/Directory/LinkDotDotSlashTextFile.txt -> ../TextFile.txt
npm WARN excluding symbolic link pkg/nuclide-fs/__mocks__/fixtures/dir/LinkDirectory -> Directory
npm WARN excluding symbolic link pkg/nuclide-fs/__mocks__/fixtures/dir/LinkDirectorySlashTextFile.txt -> Directory/TextFile.txt
npm WARN excluding symbolic link pkg/nuclide-fs/__mocks__/fixtures/dir/LinkLinkDirectory -> LinkDirectory
npm WARN excluding symbolic link pkg/nuclide-fs/__mocks__/fixtures/dir/LinkLinkDirectorySlashTextFile.txt -> LinkDirectorySlashTextFile.txt
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

Traceback (most recent call last):
  File "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/gyp/gyp_main.py", line 16, in <module>
    sys.exit(gyp.script_main())
  File "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 545, in script_main
    return main(sys.argv[1:])
  File "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 538, in main
    return gyp_main(args)
  File "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 523, in gyp_main
    generator.GenerateOutput(flat_list, targets, data, params)
  File "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 2170, in GenerateOutput
    part_of_all=qualified_target in needed_targets)
  File "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 795, in Write
    self.Pchify))
  File "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 1190, in WriteSources
    cflags = self.xcode_settings.GetCflags(configname)
  File "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 551, in GetCflags
    archs = self.GetActiveArchs(self.configname)
  File "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 420, in GetActiveArchs
    xcode_archs_default = GetXcodeArchsDefault()
  File "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 118, in GetXcodeArchsDefault
    xcode_version, _ = XcodeVersion()
  File "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 1265, in XcodeVersion
    version = re.match(r'(\d\.\d\.?\d*)', version).groups()[0]
AttributeError: 'NoneType' object has no attribute 'groups'
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/lib/configure.js:305:16)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Darwin 17.5.0
gyp ERR! command "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /private/var/folders/f8/7tv3prld4gq7mptx0w375sbm0000gn/T/apm-install-dir-11857-22107-1p8he39.bynspjh5mi/node_modules/nuclide/node_modules/runas
gyp ERR! node -v v6.9.5
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
npm WARN enoent ENOENT: no such file or directory, open '/private/var/folders/f8/7tv3prld4gq7mptx0w375sbm0000gn/T/apm-install-dir-11857-22107-1p8he39.bynspjh5mi/package.json'
npm WARN graphql-language-service-utils@1.1.0 requires a peer of graphql@^0.10.0 || ^0.11.0 || ^0.12.0 but none was installed.
npm WARN graphql-language-service@1.1.2 requires a peer of graphql@^0.6.0 || ^0.7.0 || ^0.8.0-b || ^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 but none was installed.
npm WARN relay-runtime@1.6.0 requires a peer of relay-compiler@1.6.0 but none was installed.
npm WARN graphql-language-service-types@1.2.1 requires a peer of graphql@^0.10.0 || ^0.11.0 || ^0.12.0 but none was installed.
npm WARN apm-install-dir-11857-22107-1p8he39.bynspjh5mi No description
npm WARN apm-install-dir-11857-22107-1p8he39.bynspjh5mi No repository field.
npm WARN apm-install-dir-11857-22107-1p8he39.bynspjh5mi No README data
npm WARN apm-install-dir-11857-22107-1p8he39.bynspjh5mi No license field.
npm ERR! Darwin 17.5.0
npm ERR! argv "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Users/tsunheang/.atom/.apm/.apmrc" "--userconfig" "/Users/tsunheang/.atom/.apmrc" "install" "/private/var/folders/f8/7tv3prld4gq7mptx0w375sbm0000gn/T/d-11857-22107-124qbt3.lagmrc0udi/package.tgz" "--runtime=electron" "--target=1.7.15" "--arch=x64" "--global-style"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! runas@3.1.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the runas@3.1.1 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the runas package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs runas
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls runas
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/var/folders/f8/7tv3prld4gq7mptx0w375sbm0000gn/T/apm-install-dir-11857-22107-1p8he39.bynspjh5mi/npm-debug.log
npm ERR! code 1

**Comment by drmathochist** [comment]
Sorry, I'm not quite clear on exactly what you're doing.  Are you attempting to install from scratch, or update from a previous version of Nuclide?

**Comment by jvalhondo** [comment]
Same here and same error message. 
In my case I'm trying to update from the previous v0.299.0 to v0.311.0

**Comment by jvalhondo** [comment]
It may be duplicated (not 100% sure): https://github.com/facebook/nuclide/issues/1546

**Comment by drmathochist** [comment]
@jvalhondo is it the same apm-install-dir hash?

**Comment by drmathochist** [comment]
#1546 seems to have an error with the python version:
```
gyp ERR! stack Error: Python executable "C:\[PATH]\python.EXE" is v3.6.5, which is not supported by gyp.
```

Is it possible there's a python error here too?

**Comment by drmathochist** [comment]
I'm also seeing this:
```
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```

Is it possible that XCode has updated since you last installed/upgraded Nuclide and this is now failing where it passed before?

**Comment by drmathochist** [comment]
cc @tjfryan

**Comment by liganega** [comment]
It seems to be a Python version problem. 
Until the version 0.303.0 there was no problem with installing/updating nuclide, but any trial to update it to higher version ended with the same error mentioned above. 

Having read @drmathochist 's post, I tried to symlink "python" command to "python2" instead of "python3". Then the update worked.
(You got some error messages after update, but they disappear after restarting atom.)

**Comment by liganega** [comment]
The answer by @hansonw below also indicates the problem with system Python version.

https://github.com/facebook/nuclide/issues/1541

I am wondering where there might be no problem when I revert the system Python to Python 3 
after updating nuclide.

**Comment by jvalhondo** [comment]
@drmathochist if I got you well this is the apm-install-dir hash I'm getting:
`apm-install-dir-11858-21106-jhwgzk.m7agcfpqfr`

**Comment by semmypurewal** [comment]
I think this is resolved with 0.315.0. Please re-open if you're still seeing the issue.
